kickstarter/android-oss
# EP-221: Create event for Project screen viewed

# 📲 What

- Created a new function to our `AnalyticsEvents` class that handles sending the event name and properties to the client, including which section of the project page is viewed.
- Added this function to our `ProjectViewModel` and all associated sections of the project page when their viewmodels are initialized.

# 🤔 Why

Segment Integration.

# 🛠 How

- There are 4 "context_sections" that send events in the projects flow: 
1. "overview" the project screen: Updated the event in the `ProjectViewModel`
2. "coments" the comments screen of the project: Added an event to the `CommentsViewModel`
3. "campaign" the campaign details screen of the project: Added event to the `CampaignDetailsViewModel`
4. "updates" the project updates screen: Added event to the `ProjectUpdatesViewModel`

I also had to send additional intent extras that included data required for properties. 

# 👀 See

"overview", "campaign", "updates", "comments" sections, respectively. 
![Screenshot_1613770776](https://user-images.githubusercontent.com/19390326/108564584-5bdeee00-72d1-11eb-91c0-817833bb2d39.png) ![Screenshot_1613770785](https://user-images.githubusercontent.com/19390326/108564585-5bdeee00-72d1-11eb-914e-836167101682.png) ![Screenshot_1613770793](https://user-images.githubusercontent.com/19390326/108564586-5bdeee00-72d1-11eb-8650-cdd5bccd310c.png) ![Screenshot_1613770799](https://user-images.githubusercontent.com/19390326/108564587-5bdeee00-72d1-11eb-97d2-8936c01cffc2.png)


# 📋 QA

- Select a project
- Segment should display "Page Viewed" with a property `context_section = "overview"`
- Tap on "Read more about this campaign ->" or "Campaign". This should open the project campaign details webview.
- Segment should display "Page Viewed" with a property `context_section = "campaign"`
- Tap the back arrow to return to the previous project overview screen.
- Tap "Updates". This should bring you to the project updates screen. 
- Segment should display "Page Viewed" with a property `context_section = "updates"`
- Tap the back arrow to return to the previous project overview screen.
- Tap "Comments". This should bring you to the project comments screen. 
- Segment should display "Page Viewed" with a property `context_section = "comments"`

<img width="434" alt="Screen Shot 2021-02-18 at 7 06 31 PM" src="https://user-images.githubusercontent.com/19390326/108565218-4fa76080-72d2-11eb-89ea-2406a6bfb760.png">
<img width="428" alt="Screen Shot 2021-02-18 at 7 06 38 PM" src="https://user-images.githubusercontent.com/19390326/108565220-503ff700-72d2-11eb-97e9-03c8a1d3b34c.png">
<img width="428" alt="Screen Shot 2021-02-18 at 7 06 50 PM" src="https://user-images.githubusercontent.com/19390326/108565221-503ff700-72d2-11eb-8031-5d9262e7f30f.png">
<img width="431" alt="Screen Shot 2021-02-18 at 8 25 11 PM" src="https://user-images.githubusercontent.com/19390326/108565222-503ff700-72d2-11eb-80d1-8345b1a43fe7.png">

# Story 📖

[EP-221: Android Page Viewed (project)](https://kickstarter.atlassian.net/browse/EP-221)
